x-pack/plugins/encrypted_saved_objects/server/create_migration.ts

```diff
@@ -80,20 +92,40 @@ export const getCreateMigration = (
     const encryptDescriptor = { id, type, namespace: encryptedDoc.namespace };
 
     // decrypt the attributes using the input type definition
-    // then migrate the document
-    // then encrypt the attributes using the migration type definition
-    return mapAttributes(
-      migration(
-        mapAttributes(encryptedDoc, (inputAttributes) =>
-          inputService.decryptAttributesSync<any>(decryptDescriptor, inputAttributes, {
+    // if an error occurs during decryption, use the shouldMigrateIfDecryptionFails flag
+    // to determine whether to throw the error or continue the migration
+    // if we are continuing the migration, strip encrypted attributes from the document using stripOrDecryptAttributesSync
+    const documentToMigrate = mapAttributes(encryptedDoc, (inputAttributes) => {
+      try {
+        const decryptedAttributes = inputService.decryptAttributesSync<any>(
+          decryptDescriptor,
+          inputAttributes,
+          { convertToMultiNamespaceType }
+        );
+        return decryptedAttributes;
+      } catch (err) {
+        if (!shouldMigrateIfDecryptionFails || !(err instanceof EncryptionError)) {
+          throw err;
+        }
+
+        context.log.warn(
+          `Decryption failed for encrypted Saved Object "${encryptedDoc.id}" of type "${encryptedDoc.type}" with error: ${err.message}. Migration will be applied to the original encrypted document but this may cause decryption errors later on.`
+        );
+        const { attributes: strippedAttributes } = inputService.stripOrDecryptAttributesSync<any>(
+          decryptDescriptor,
+          inputAttributes,
+          {
             convertToMultiNamespaceType,
-          })
-        ),
-        context
-      ),
-      (migratedAttributes) =>
-        migratedService.encryptAttributesSync<any>(encryptDescriptor, migratedAttributes)
-    );
+          }
+        );
+        return strippedAttributes;
```

Comment:
__optional nit:__ (a little bit easier to read since we know `strippedAttributes` is not used anywhere else except `return`)
```ts
        return inputService.stripOrDecryptAttributesSync<any>(decryptDescriptor, inputAttributes, {
          convertToMultiNamespaceType,
        }).attributes;
```

Reply:
Updated in https://github.com/elastic/kibana/pull/105968/commits/1365d05c60aeae8c4bd64962cfa6f93293dc4e0d